shims/spark35/src/main/scala/org/apache/spark/sql/execution/adaptive/GlutenCostEvaluator.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.execution.adaptive
+
+import org.apache.spark.sql.catalyst.SQLConfHelper
+import org.apache.spark.sql.errors.QueryExecutionErrors
+import org.apache.spark.sql.execution.SparkPlan
+import org.apache.spark.sql.internal.SQLConf
+
+case class GlutenCost(value: Long, planId: Int) extends Cost {
+  override def compare(that: Cost): Int = that match {
+    case GlutenCost(thatValue, thatId) =>
+      if (value < thatValue || (value == thatValue && planId != thatId)) -1
+      else if (value > thatValue) 1
+      else 0
+    case _ =>
+      throw QueryExecutionErrors.cannotCompareCostWithTargetCostError(that.toString)
+  }
+}
+
+/** This [[CostEvaluator]] is to force use the new physical plan when cost is equal. */
+case class GlutenCostEvaluator() extends CostEvaluator with SQLConfHelper {
+  override def evaluateCost(plan: SparkPlan): Cost = {
+    val forceOptimizeSkewedJoin = conf.getConf(SQLConf.ADAPTIVE_FORCE_OPTIMIZE_SKEWED_JOIN)
+    val simpleCost = SimpleCostEvaluator(forceOptimizeSkewedJoin)
+      .evaluateCost(plan)
+      .asInstanceOf[SimpleCost]
+    GlutenCost(simpleCost.value, plan.id)
```

Comment:
Can we add one more config to crontrol this feature ? if disable then do nothing.

Reply:
How about add GlutenConfig `spark.gluten.sql.execution.adaptive.costEvaluator.enabled`, if true, use `GlutenCostEvaluator`, else follow the configuration `org.apache.spark.sql.execution.adaptive.GlutenCostEvaluator`?

Reply:
Sounds good, besides, we'd better to respect spark.gluten.enabled even costEvaluator is enabled. People may tune spark.gluten.enabled off at runtime.